alexandria/core/management/commands/generate_content_vectors.py

```diff
@@ -17,13 +17,19 @@ def handle(self, *args, **options):
                 )
             )
 
+        # disable additional features to focus on vectorization
+        checksum = settings.ALEXANDRIA_ENABLE_CHECKSUM
+        thumbnail = settings.ALEXANDRIA_ENABLE_THUMBNAIL_GENERATION
+        settings.ALEXANDRIA_ENABLE_CHECKSUM = False
+        settings.ALEXANDRIA_ENABLE_THUMBNAIL_GENERATION = False
+
         failed_files = []
-        for file in tqdm(
-            File.objects.filter(variant="original", content_vector__isnull=True)
-        ):
+        query = File.objects.filter(variant="original", content_vector__isnull=True)
+        # iterate over files in batches to prevent memory exhaustion
+        for file in tqdm(query.iterator(50), "Generating vectors", query.count()):
```

Comment:
Bottleneck seems to be tika, tried implementing this with multithreading but it always timed out when parsing the content with tika.
Speed is about 2-3 files per second.

Reply:
Might not even need the memory check anymore, ran without memory problems even after 1500 files.